File: Vanilla/sources/zdoc/media/lua/shared/Library/BeardStyle.lua
---@class BeardStyle : zombie.core.skinnedmodel.population.BeardStyle
---@field public name String
---@field public model String
---@field public texture String
---@field public level int
---@field public trimChoices ArrayList|Unknown
---@field public growReference boolean
BeardStyle = {}

---@public
---@return int
function BeardStyle:getLevel() end

---@public
---@return boolean
function BeardStyle:isValid() end

---@public
---@return String
function BeardStyle:getName() end

---@public
---@return ArrayList|Unknown
function BeardStyle:getTrimChoices() end

File: Vanilla/sources/zomboid/zombie/core/skinnedmodel/population/BeardStyle.java
package zombie.core.skinnedmodel.population;

import java.util.ArrayList;
import zombie.util.StringUtils;

public class BeardStyle {
   public String name = "";
   public String model;
   public String texture = "F_Hair_White";
   public int level = 0;
   public ArrayList trimChoices = new ArrayList();
   public boolean growReference = false;

   public boolean isValid() {
      return !StringUtils.isNullOrWhitespace(this.model) && !StringUtils.isNullOrWhitespace(this.texture);
   }

   public int getLevel() {
      return this.level;
   }

   public String getName() {
      return this.name;
   }

   public ArrayList getTrimChoices() {
      return this.trimChoices;
   }
}
